bsp4j/src/main/java/ch/epfl/scala/bsp4j/RustBuildServer.java

```diff
@@ -0,0 +1,15 @@
+package ch.epfl.scala.bsp4j;
+
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+
+public interface RustBuildServer {
+    @JsonRequest("buildTarget/rustOptions")
+    CompletableFuture<RustOptionsResult> buildTargetRustOptions(RustOptionsParams params);
+
+    @JsonRequest("buildTarget/rustMetadata")
+    CompletableFuture<RustMetadataResult> rustMetadata();
```

Comment:
I would suggest adding `RustMetadataParams` even if they would be empty now. If in the future we need to pass something, we have an already existing way.

Reply:
Currently, `metadata` has only one version, but as there is already an opportunity to pick a version, we should be aware.
https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_output_metadata.rs#L28

Reply:
Done